### DISCUSSION TITLE
[Feature reques] add multi-line parameters support

### DISCUSSION THREAD

**Discussion by jheroy** [knqyf263/pet#300] [General]
Current parameters only support one line, press enter will exec a snippet. Use '\n' or shift+enter to support multi-line parameters.

**Comment by RamiAwar** [comment]


**Reply by jheroy** [reply]
This is not the functionality I want. What I want is for the input box to be able to wrap and accept multiple lines when executing a snippet with parameters, instead of doing this when generating a new snippet.

**Reply by RamiAwar** [reply]
Ah sorry misread as multi-line snippets.

**Comment by RamiAwar** [comment]
What's the use-case for this? Can you give me an example?

I'm finding it hard to think of one myself! @jheroy

**Comment by jheroy** [comment]
I often need to execute the same SQL statements, which are multiline,
across different databases in a testing environment.

Rami Awar ***@***.***> 于 2024年5月3日周五 00:41写道：

**Reply by RamiAwar** [reply]
I see. Can you test out something for me? If you enter your sql statements into a parameter with newlines separated by `\n`, would the command work?

Something like this:
```
❯ pet new
Command> echo "<param=multi line command \n goes here>"
Description> multiline param
❯ echo "multi line command \n goes here"
multi line command
 goes here
 ```

**Reply by jheroy** [reply]
It's not work, just output string "\n".

**Reply by jheroy** [reply]
```bash
> pet new 
Command> echo "<param=multi line command \n goes here>"
Description> multiline param

> pet exec 
> echo "multi line command \n goes here"
multi line command \n goes here
```

**Reply by jheroy** [reply]
This is the snippet:
```ini
[[snippets]]
  description = "multiline param"
  command = "echo \"<param=multi line command \\n goes here>\""
  output = ""
```
I change it to :
```ini
[[snippets]]
  description = "multiline param"
  command = "echo \"<param=multi line command \n goes here>\""
  output = ""
```
exec result:
```bash
> pet exec 
> echo "multi line command  goes here"
multi line command  goes here
```

**Reply by RamiAwar** [reply]
Are you on the latest version? Try doing a `brew update && brew upgrade pet`

**Reply by jheroy** [reply]
I manually downloaded the latest version from the release page, the running result is still the same as before. by the way, I'm using the Arch Linux system.
```bash
> pet version
pet version 0.8.4
```

**Reply by RamiAwar** [reply]
Can you try it without `pet exec`? Instead try using the pet search shortcut that would paste the command to your terminal and then you execute it manually.

In the docs its setup to be ctrl+S for example. 

That's the way I recommend using Pet
